### DISCUSSION TITLE
Hardware Buttons pressed=TRUE?

### DISCUSSION THREAD

**Discussion by kptkip** [blakadder/nspanel#26] [Q&A]
Hi Folks,

can I get different information via MQTT about the relays being switched on/off either by software or the hardware buttons?

**Comment by joBr99** [comment]
This isn't specifc to this berry driver, please see Tasmota documentation, the buttons are directly connected to the esp32/tasmota.

https://tasmota.github.io/docs/Rules/#send-mqtt-message-on-button-press

**Reply by kptkip** [reply]
Thanks for the hint.
Unfortunately it doesn't work. to use button rules you have to set `ButtonTopic 0`. 
But then the link between the button and the relay is cut.

But thanks for the hint on tasmota as the general source. I have to search a little bit furthermore.

**Reply by joBr99** [reply]
This is true, you can set the relay state in your rule aswell. (But then you could also check the relay state in the first place without any rules)

**Reply by kptkip** [reply]
I realize another problem:
The NSPanel has _buttons_ no _switches_. So there is only a momentary state of button pressed and no continuous one. This could also lead to problems.
The only solution could be persisting this information on the smarthome system (in my case ioBroker) and calculate there further logic.

**Reply by joBr99** [reply]
You can toggle the state of the relay with rules and also use variables in rules.
https://tasmota.github.io/docs/Rules/#setting-variables
